<?xml version="1.0" encoding="UTF-8" ?>
<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom" xmlns:dc="http://purl.org/dc/elements/1.1/">
  <channel>
    <title>זוהר argv - הערת mypy על ערכי החזרה לאחר הוספת טיפול בשגיאות</title>
    <link>https://forums.pythonic.guru/t/argv-mypy/7972</link>
    <description>חשבתי לתרגל קצת את הmypy בתרגיל הזה והכל עבד יפה. 
אולם שבחרתי לנסות להוסיף טיפול בשגיאות בפורמט של try .. except  
mypy זורק לי את ההערה הבאה:
```
mypy error missing return statement
```

הקוד שלי מחזיר int או str  ולאחר הוספת הטיפול בשגיאות בגדול הוא יכול גם להחזיר None.
ערך ההחזרה של הפונקציה שלי הוא:
```
-&gt; Optional[Union[int, str]]
```

לפני שהוספתי את הטיפול בשגיאות ואיתו גם את Optional  כמובן אז גם תחת --strict  לא היו שגיאות.</description>
    <language>he</language>
    <lastBuildDate>Wed, 26 Aug 2020 08:03:10 +0000</lastBuildDate>
    <category>שבוע 10</category>
    <atom:link href="https://forums.pythonic.guru/t/argv-mypy/7972.rss" rel="self" type="application/rss+xml" />
      <item>
        <title>זוהר argv - הערת mypy על ערכי החזרה לאחר הוספת טיפול בשגיאות</title>
        <dc:creator><![CDATA[aviadamar]]></dc:creator>
        <description><![CDATA[
            <p>אמת. התרגלתי לרשום בלי <img src="https://forums.pythonic.guru/images/emoji/twitter/slight_smile.png?v=9" title=":slight_smile:" class="emoji" alt=":slight_smile:"> תודה.</p>
<p>לידיעת הכלל: גם שרושמים רק return ולא return None  גם חוזרת אותה השגיאה.</p>
          <p><a href="https://forums.pythonic.guru/t/argv-mypy/7972/3">קריאת הנושא במלואו</a></p>
        ]]></description>
        <link>https://forums.pythonic.guru/t/argv-mypy/7972/3</link>
        <pubDate>Wed, 26 Aug 2020 08:03:10 +0000</pubDate>
        <guid isPermaLink="false">forums.pythonic.guru-post-7972-3</guid>
        <source url="https://forums.pythonic.guru/t/argv-mypy/7972.rss">זוהר argv - הערת mypy על ערכי החזרה לאחר הוספת טיפול בשגיאות</source>
      </item>
      <item>
        <title>זוהר argv - הערת mypy על ערכי החזרה לאחר הוספת טיפול בשגיאות</title>
        <dc:creator><![CDATA[Yam]]></dc:creator>
        <description><![CDATA[
            <p>נשמע שיש מצב בפונקציה שהיא לא מגיעה ל־<code>return</code>.<br>
אם אתה ב־<code>--strict</code>, mypy מבקש שתרשום במפורש <code>return None</code></p>
          <p><a href="https://forums.pythonic.guru/t/argv-mypy/7972/2">קריאת הנושא במלואו</a></p>
        ]]></description>
        <link>https://forums.pythonic.guru/t/argv-mypy/7972/2</link>
        <pubDate>Wed, 26 Aug 2020 08:00:06 +0000</pubDate>
        <guid isPermaLink="false">forums.pythonic.guru-post-7972-2</guid>
        <source url="https://forums.pythonic.guru/t/argv-mypy/7972.rss">זוהר argv - הערת mypy על ערכי החזרה לאחר הוספת טיפול בשגיאות</source>
      </item>
      <item>
        <title>זוהר argv - הערת mypy על ערכי החזרה לאחר הוספת טיפול בשגיאות</title>
        <dc:creator><![CDATA[aviadamar]]></dc:creator>
        <description><![CDATA[
            <p>חשבתי לתרגל קצת את הmypy בתרגיל הזה והכל עבד יפה.<br>
אולם שבחרתי לנסות להוסיף טיפול בשגיאות בפורמט של try … except<br>
mypy זורק לי את ההערה הבאה:</p>
<pre><code class="lang-auto">mypy error missing return statement
</code></pre>
<p>הקוד שלי מחזיר int או str  ולאחר הוספת הטיפול בשגיאות בגדול הוא יכול גם להחזיר None.<br>
ערך ההחזרה של הפונקציה שלי הוא:</p>
<pre><code class="lang-auto">-&gt; Optional[Union[int, str]]
</code></pre>
<p>לפני שהוספתי את הטיפול בשגיאות ואיתו גם את Optional  כמובן אז גם תחת --strict  לא היו שגיאות.</p>
          <p><a href="https://forums.pythonic.guru/t/argv-mypy/7972/1">קריאת הנושא במלואו</a></p>
        ]]></description>
        <link>https://forums.pythonic.guru/t/argv-mypy/7972/1</link>
        <pubDate>Wed, 26 Aug 2020 07:31:49 +0000</pubDate>
        <guid isPermaLink="false">forums.pythonic.guru-post-7972-1</guid>
        <source url="https://forums.pythonic.guru/t/argv-mypy/7972.rss">זוהר argv - הערת mypy על ערכי החזרה לאחר הוספת טיפול בשגיאות</source>
      </item>
  </channel>
</rss>
